src/glum/_glm.py

```diff
@@ -1552,6 +1625,53 @@ def score(
         dev_null = family.deviance(y, y_mean, sample_weight=sample_weight)
         return 1.0 - dev / dev_null
 
+    def _compute_information_criteria(self):
+        """
+        Computes and stores the model's degrees of freedom, and the 'aic',
+        'aicc' and 'bic' information criterion for the model.
+
+        References
+        ----------
+        [1] Burnham KP, Anderson KR (2002). Model Selection and Multimodel
+        Inference; Springer New York.
+        """
+
+        # we require that the log_likelihood be definied
+        if not isinstance(
+            self.family_instance,
+            (
+                BinomialDistribution,
+                GammaDistribution,
+                NormalDistribution,
+                PoissonDistribution,
+            ),
+        ):
+            return False
+
+        ddof = np.sum(np.abs(self.coef_) > np.finfo(self.coef_.dtype).eps)
+
+        X = self._info_criteria["X"]
+        y = self._info_criteria["y"]
```

Comment:
Why not just take `X` and `y` as arguments?

Reply:
See answer above. These are *in sample* metrics that are used to approximate out of sample error (they do this by penalising the complexity of the model). So I wanted to make this method run when the `fit` method was called, thereby forcing that in-sample data be used.